### ISSUE TITLE
Updating chart without full refresh/flicker

### ISSUE THREAD

**Issue by andreialecu** [donothingloop/ng2-charts-x#3]
I noticed that if I just mutate the array passed to `[data]` the chart won't update. It seems like `[data]` needs to be fully replaced with a new instance of an array on updates in order for it to be picked up. 

This is fine, and I see why it works this way. (a method you could call on the chart instance to force an update would be nice though)

However, there's a bug with the labels implementation. Simply resetting it to the same labels will also refresh the chart. So if you repeatedly set a `labels` variable to `["A", "B"]` so that a new array is passed every time, the chart will refresh on every update.

An easy repro:

```javascript
labels: [];

ngOnInit() {
  setInterval(() => { this.labels = ["A", "B"]; }, 1000);
}
}
```

If you have a chart's `[labels]` bound to this variable it will refresh every second.

The problem is here: 
https://github.com/donothingloop/ng2-charts-x/blob/9424606b44d61d4eebd4dc773d1f24d5154b21b3/src/charts/charts.directive.ts#L72

An easy fix would be to deep compare the `previousValue` and `currentValue` of `labels` before calling `this.refresh()`

**Comment by donothingloop** [comment]
The same thing would be true for options, chartType, colors and legend.
I don't know if the current behavior is really that bad.

Why are you setting the labels array continuously to the same values?

**Comment by andreialecu** [comment]
It's just how the code is set up in this particular case. It queries the database for the labels/polls and then updates the bindings. So they're going to be new objects all the time. 

I did work around it by updating the objects in place, but it's a hack that could be avoided.

**Comment by donothingloop** [comment]
You could do the deep equal comparison in your code before assigning the arrays. The current behaviour is the more angular-standard way of doing this I think.

Also there might be a performance impact for certain applications if this is checked by the library.

**Comment by andreialecu** [comment]
Yes but this is exactly the opposite of how `[data]` works. There are checks in place there to refresh the chart only if needed.

See:
https://github.com/donothingloop/ng2-charts-x/blob/master/src/charts/charts.directive.ts#L82

`this.refresh()` is only called if series have changed, not the data itself. `this.refresh()` completely rebuilds the chart, so the same logic could be applied to the `labels` check above.

To recap:

1) Just updating `labels` (in place, mutating the same array) without updating `data` has no effect. Labels will not change.
2) Mutating `labels` while replacing `data` makes the chart smoothly update.
3) Replacing `labels` and mutating `data` will refresh the chart, update the labels, but not update the data.

The only way to make all of these scenarios work would be to implement `ngDoCheck` and do a deep compare.

**Comment by andreialecu** [comment]
As an example of a possible implementation, `ngStyle` in @angular/core needs to work in a similar way, [`KeyValueDiffers`](https://angular.io/api/core/KeyValueDiffers) might help.

Source:
https://github.com/angular/angular/blob/1c85e9958874402b404909056bec64a13763984d/modules/@angular/common/src/directives/ng_style.ts
